src/metric/event.rs

```diff
@@ -41,6 +42,8 @@ pub enum Event {
         encoding: Encoding,
         /// Encoded payload.
         bytes: Vec<u8>,
+        /// Header values used by some sinks to send metadata
```

Comment:
There are a lot of `Event` instances that get made inside cernan. I can't recall off the top of my head how the enum field variants we see here affect size. That is, does `Raw` contribute more to the total size of `Event` than `Telemetry(Telemetry)`? I can't recall off the top of my head. Worth doing a size_of to find out. 

The smaller `Event` is the better.